src/compiler/crystal/exception.cr

```diff
@@ -94,4 +99,194 @@ module Crystal
       end
     end
   end
+
+  module ErrorFormat
```

Comment:
Is this module needed? Can't we just put the methods in `Crystal::Exception`?

Reply:
Only two Exception classes utilise these methods: `TypeException` and `SyntaxException`. Therefore, I felt that adding them to `Crystal::Exception` would bloat the instances that don't use these methods.

Happy to move it if that would result in a better outcome.

Reply:
The end result is actually more bloat: methods from modules are typed once for each included type, and also it generates duplicate code. This isn't something that you should or need know... but that's the current state of Crystal and so the less bloat in the compiler's generated code, the better, I think.

In any case, it's probably unnoticeable.